lib/core/include/scipp/core/histogram.h

```diff
@@ -33,6 +33,10 @@ Index get_bin(const T &x, const Edges &edges, const Params &params) {
   // integer overflow when converting the "bin" computation result to `Index`.
   if (x < edges.front() || x >= edges.back())
     return -1;
+  // If x is not finite we also return early as it cannot be in any bin.
+  if constexpr (!std::is_same_v<T, time_point>)
+    if (!std::isfinite(x))
```

Comment:
Why do we need to use `isfinite` and not `isnan`? Infinite values should work perfectly fine with `<` and `>=`, and it could even be value to have a bin edge at `-inf` to include those values (would not work though for the open bound at the right).

Reply:
True, I'll just use `isnan`.